templates/terraform/iam/iam_attributes.go.erb

```diff
@@ -1,13 +1,13 @@
-<% example = object.examples.reject(&:skip_test)
+<% example = object.examples
          .reject { |e| @api.version_obj_or_closest(version) < @api.version_obj_or_closest(e.min_version) }
          .first -%>
 <% parent_resource_last_param_name = extract_identifiers(object.self_link_url).last -%>
-<% parent_resource_type_type = object.iam_policy.parent_resource_type || terraform_name -%>
+<% parent_resource_type_type = object.iam_policy.parent_resource_type || terraform_name.delete_suffix("_iam") -%>
```

Comment:
Why do we need to delete the `_iam` suffix? Shouldn't the parent never be an IAM resource?

Reply:
Not sure what you mean, but if you don't specify `parent_resource_type` here, `terraform_name` as defined in the parent template is "#{resource_ns}_iam` as the prefix to the IAM resource names and it'll generate `${google_cloudfunctions_function_iam.function.cloud_function}"`

Reply:
That makes sense. I was confusing this with the specified `parent_resource_type`

Reply:
ahhh so i think the real issue is that terraform_name in iam_test_file.go.erb doesn't have the suffix, whereas in the website file, it does. I think i'd prefer if we were stricter and explicitly define the local variables we pass to the templates, even if it makes the parent templates a bit uglier